doc/going-further/extending-castor/events.md

```diff
@@ -6,22 +6,50 @@ application lifecycle.
 
 ## Registering a listener
 
-You can register a listener inside your Castor project by using the
-`Castor\Attribute\AsListener` attribute. This attribute allows you to specify
-the targeted event and the priority of this listener.
+You can register a listener inside your Castor project by using the `Castor\Attribute\AsListener` attribute as a **function** (not a class method). This attribute allows you to specify the targeted event and the priority of this listener.
 
+**Important:** Listener functions must be defined in a file that is imported in your `castor.php` file using the `import()` function.
+
+### Example
 ```php
+<?php
+// file: src/EventListener/CastorListener.php
+
+use Castor\Attribute\AsListener;
 use Castor\Event\AfterExecuteTaskEvent;
-use Castor\Event\FunctionsResolvedEvent;
+use Castor\Event\ContextCreatedEvent;
 
-#[AsListener(event: AfterExecuteTaskEvent::class)]
-#[AsListener(event: FunctionsResolvedEvent::class, priority: 1)]
-function my_event_listener(AfterExecuteTaskEvent|FunctionsResolvedEvent $event): void
+#[AsListener(AfterExecuteTaskEvent::class)]
+#[AsListener(ContextCreatedEvent::class)]
+function my_event_listener(AfterExecuteTaskEvent|ContextCreatedEvent $event): void
```

Comment:
You go too hard at the begining. Let's display a simple example, with only one event.

Then, later in the article, we can show how to listen multiple events